src/pages/MovieListPage/Page/MovieListPage.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+
+import {
+  SearchForm,
+  SortControl,
+  GenreSelect,
+  MovieTile,
+  MovieDetails,
+  Genre,
+  Movie,
+  SortOption,
+} from "../../../components";
+
+import { genres } from "../../../components/GenreSelect/Component/GenreSelect";
+import styles from "../Styles/MovieListPage.module.css";
+import useMovieQuery from "../../../hooks/useMovieQuery";
+
+const MovieListPage: React.FC = () => {
+  const [searchQuery, setSearchQuery] = useState<string>("");
+  const [selectedGenre, setSelectedGenre] = useState<Genre>(genres[0]);
+  const [selectedSort, setSelectedSort] = useState<SortOption>("release_date");
+  const [selectedMovie, setSelectedMovie] = useState<Movie | null>(null);
+
+  const { isLoading, data } = useMovieQuery(
+    searchQuery,
+    selectedGenre,
+    selectedSort
+  );
+
+  const onSearch = (searchText: string) => {
+    setSearchQuery(searchText);
+  };
+
+  const handleSortChange = (newSort: SortOption) => {
+    setSelectedSort(newSort);
+  };
+
+  const handleGenreSelect = (genre: Genre) => {
+    setSelectedGenre(genre);
+  };
+
+  function onMovieClick(movie: Movie | null): void {
+    setSelectedMovie(movie);
+  }
+
+  const renderMovieTiles = (movies: Movie[] | undefined) => {
+    return movies?.map((movie) => (
+      <MovieTile
+        key={movie.id}
+        movie={movie}
+        onClick={() => onMovieClick(movie)}
+        onEdit={() => onMovieClick(movie)}
```

Comment:
onEdit and onDelete - should not be here. What is the reason for them to 'live' here and be passed inyo a Tile as pops?

Reply:
This is for Edit and Delete menu on every tile